File: Common/TextWriterWrapperTemplate.vb
Public Class TextWriterWrapper
    Inherits IO.TextWriter

    Public Sub New(ByVal TextWriter As IO.TextWriter)
        Me.Base = TextWriter
    End Sub

#Region "Writes Group"
    Public Overrides Sub Write(value As String)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(buffer() As Char, index As Integer, count As Integer)
        Me.Base.Write(buffer, index, count)
    End Sub

    Public Overrides Sub Write(value As Char)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(buffer() As Char)
        Me.Base.Write(buffer)
    End Sub

    Public Overrides Sub Write(format As String, arg0 As Object)
        Me.Base.Write(format, arg0)
    End Sub

    Public Overrides Sub Write(format As String, arg0 As Object, arg1 As Object)
        Me.Base.Write(format, arg0, arg1)
    End Sub

    Public Overrides Sub Write(format As String, arg0 As Object, arg1 As Object, arg2 As Object)
        Me.Base.Write(format, arg0, arg1, arg2)
    End Sub

    Public Overrides Sub Write(format As String, ParamArray arg() As Object)
        Me.Base.Write(format, arg)
    End Sub

    Public Overrides Sub Write(value As Boolean)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(value As Decimal)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(value As Double)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(value As Integer)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(value As Long)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(value As Object)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(value As Single)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(value As UInteger)
        Me.Base.Write(value)
    End Sub

    Public Overrides Sub Write(value As ULong)
        Me.Base.Write(value)
    End Sub
#End Region

#Region "WriteLines Group"
    Public Overrides Sub WriteLine()
        Me.Base.WriteLine()
    End Sub

    Public Overrides Sub WriteLine(buffer() As Char)
        Me.Base.WriteLine(buffer)
    End Sub

    Public Overrides Sub WriteLine(buffer() As Char, index As Integer, count As Integer)
        Me.Base.WriteLine(buffer, index, count)
    End Sub

    Public Overrides Sub WriteLine(format As String, arg0 As Object)
        Me.Base.WriteLine(format, arg0)
    End Sub

    Public Overrides Sub WriteLine(format As String, arg0 As Object, arg1 As Object)
        Me.Base.WriteLine(format, arg0, arg1)
    End Sub

    Public Overrides Sub WriteLine(format As String, arg0 As Object, arg1 As Object, arg2 As Object)
        Me.Base.WriteLine(format, arg0, arg1, arg2)
    End Sub

    Public Overrides Sub WriteLine(format As String, ParamArray arg() As Object)
        Me.Base.WriteLine(format, arg)
    End Sub

    Public Overrides Sub WriteLine(value As Boolean)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As Char)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As Decimal)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As Double)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As Integer)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As Long)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As Object)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As Single)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As String)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As UInteger)
        Me.Base.WriteLine(value)
    End Sub

    Public Overrides Sub WriteLine(value As ULong)
        Me.Base.WriteLine(value)
    End Sub
#End Region

#Region "Asyncs Group"
    Public Overrides Function WriteAsync(buffer() As Char, index As Integer, count As Integer) As Task
        Return Me.Base.WriteAsync(buffer, index, count)
    End Function

    Public Overrides Function WriteAsync(value As Char) As Task
        Return Me.Base.WriteAsync(value)
    End Function

    Public Overrides Function WriteAsync(value As String) As Task
        Return Me.Base.WriteAsync(value)
    End Function

    Public Overrides Function WriteLineAsync() As Task
        Return Me.Base.WriteLineAsync()
    End Function

    Public Overrides Function WriteLineAsync(buffer() As Char, index As Integer, count As Integer) As Task
        Return Me.Base.WriteLineAsync(buffer, index, count)
    End Function

    Public Overrides Function WriteLineAsync(value As Char) As Task
        Return Me.Base.WriteLineAsync(value)
    End Function

    Public Overrides Function WriteLineAsync(value As String) As Task
        Return Me.Base.WriteLineAsync(value)
    End Function
#End Region

    Public Overrides Sub Close()
        Me.Base.Close()
    End Sub

    Public Overrides Function CreateObjRef(requestedType As Type) As Runtime.Remoting.ObjRef
        Return Me.Base.CreateObjRef(requestedType)
    End Function

    Protected Overrides Sub Dispose(disposing As Boolean)
        Me.Base.Dispose()
    End Sub

    Public Overrides Sub Flush()
        Me.Base.Flush()
    End Sub

    Public Overrides Function FlushAsync() As Task
        Return Me.Base.FlushAsync()
    End Function

    Public Overrides Function InitializeLifetimeService() As Object
        Return Me.Base.InitializeLifetimeService()
    End Function

    Public Overrides Function ToString() As String
        Return String.Concat(NameOf(TextWriterWrapper), "{", Me.Base.ToString(), "}")
    End Function

    Public Overrides ReadOnly Property Encoding As Text.Encoding
        Get
            Return Me.Base.Encoding
        End Get
    End Property

    Public Overrides Property NewLine As String
        Get
            Return Me.Base.NewLine
        End Get
        Set(value As String)
            Me.Base.NewLine = value
        End Set
    End Property

    Public Overrides ReadOnly Property FormatProvider As IFormatProvider
        Get
            Return Me.Base.FormatProvider
        End Get
    End Property

    Private ReadOnly Base As IO.TextWriter

End Class

File: Common/TextWriterWrapperTemplate.cs
using System;
using System.Threading.Tasks;

namespace Ks.Common
{
    public class TextWriterWrapper : System.IO.TextWriter
    {
        public TextWriterWrapper(System.IO.TextWriter TextWriter)
        {
            this.Base = TextWriter;
        }

        public override void Write(string value)
        {
            this.Base.Write(value);
        }

        public override void Write(char[] buffer, int index, int count)
        {
            this.Base.Write(buffer, index, count);
        }

        public override void Write(char value)
        {
            this.Base.Write(value);
        }

        public override void Write(char[] buffer)
        {
            this.Base.Write(buffer);
        }

        public override void Write(string format, object arg0)
        {
            this.Base.Write(format, arg0);
        }

        public override void Write(string format, object arg0, object arg1)
        {
            this.Base.Write(format, arg0, arg1);
        }

        public override void Write(string format, object arg0, object arg1, object arg2)
        {
            this.Base.Write(format, arg0, arg1, arg2);
        }

        public override void Write(string format, params object[] arg)
        {
            this.Base.Write(format, arg);
        }

        public override void Write(bool value)
        {
            this.Base.Write(value);
        }

        public override void Write(decimal value)
        {
            this.Base.Write(value);
        }

        public override void Write(double value)
        {
            this.Base.Write(value);
        }

        public override void Write(int value)
        {
            this.Base.Write(value);
        }

        public override void Write(long value)
        {
            this.Base.Write(value);
        }

        public override void Write(object value)
        {
            this.Base.Write(value);
        }

        public override void Write(float value)
        {
            this.Base.Write(value);
        }

        public override void Write(uint value)
        {
            this.Base.Write(value);
        }

        public override void Write(ulong value)
        {
            this.Base.Write(value);
        }

        public override void WriteLine()
        {
            this.Base.WriteLine();
        }

        public override void WriteLine(char[] buffer)
        {
            this.Base.WriteLine(buffer);
        }

        public override void WriteLine(char[] buffer, int index, int count)
        {
            this.Base.WriteLine(buffer, index, count);
        }

        public override void WriteLine(string format, object arg0)
        {
            this.Base.WriteLine(format, arg0);
        }

        public override void WriteLine(string format, object arg0, object arg1)
        {
            this.Base.WriteLine(format, arg0, arg1);
        }

        public override void WriteLine(string format, object arg0, object arg1, object arg2)
        {
            this.Base.WriteLine(format, arg0, arg1, arg2);
        }

        public override void WriteLine(string format, params object[] arg)
        {
            this.Base.WriteLine(format, arg);
        }

        public override void WriteLine(bool value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(char value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(decimal value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(double value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(int value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(long value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(object value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(float value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(string value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(uint value)
        {
            this.Base.WriteLine(value);
        }

        public override void WriteLine(ulong value)
        {
            this.Base.WriteLine(value);
        }

        public override Task WriteAsync(char[] buffer, int index, int count)
        {
            return this.Base.WriteAsync(buffer, index, count);
        }

        public override Task WriteAsync(char value)
        {
            return this.Base.WriteAsync(value);
        }

        public override Task WriteAsync(string value)
        {
            return this.Base.WriteAsync(value);
        }

        public override Task WriteLineAsync()
        {
            return this.Base.WriteLineAsync();
        }

        public override Task WriteLineAsync(char[] buffer, int index, int count)
        {
            return this.Base.WriteLineAsync(buffer, index, count);
        }

        public override Task WriteLineAsync(char value)
        {
            return this.Base.WriteLineAsync(value);
        }

        public override Task WriteLineAsync(string value)
        {
            return this.Base.WriteLineAsync(value);
        }

        public override void Close()
        {
            this.Base.Close();
        }

        public override System.Runtime.Remoting.ObjRef CreateObjRef(Type requestedType)
        {
            return this.Base.CreateObjRef(requestedType);
        }

        protected override void Dispose(bool disposing)
        {
            this.Base.Dispose();
        }

        public override void Flush()
        {
            this.Base.Flush();
        }

        public override Task FlushAsync()
        {
            return this.Base.FlushAsync();
        }

        public override object InitializeLifetimeService()
        {
            return this.Base.InitializeLifetimeService();
        }

        public override string ToString()
        {
            return string.Concat(nameof(TextWriterWrapper), "{", this.Base.ToString(), "}");
        }

        public override System.Text.Encoding Encoding
        {
            get
            {
                return this.Base.Encoding;
            }
        }

        public override string NewLine
        {
            get
            {
                return this.Base.NewLine;
            }
            set
            {
                this.Base.NewLine = value;
            }
        }

        public override IFormatProvider FormatProvider
        {
            get
            {
                return this.Base.FormatProvider;
            }
        }

        private readonly System.IO.TextWriter Base;
    }
}
